### ISSUE TITLE
ocaml-freestanding fails to build on OpenBSD

### ISSUE THREAD

**Issue by adamsteen** [mirage/ocaml-freestanding#70]
Hi All

I am not sure if anyone else uses OpenBSD, but it currently fails, I will eventually get back to fix this, but wanted it known by a larger audience.

Sorry
Adam

```
[ERROR] The compilation of ocaml-freestanding failed at "/usr/local/bin/gmake".

#=== ERROR while compiling ocaml-freestanding.0.4.7 ===========================#
# context     2.0.5 | openbsd/x86_64 | ocaml-system.4.09.0 | https://opam.ocaml.org#c2df68ef
# path        ~/.opam/default/.opam-switch/build/ocaml-freestanding.0.4.7
# command     /usr/local/bin/gmake
# exit-code   2
# env-file    ~/.opam/log/ocaml-freestanding-42466-a4623a.env
# output-file ~/.opam/log/ocaml-freestanding-42466-a4623a.out
### output ###
# [...]
# In file included from ./caml/io.h:33:
# In file included from /home/asteen/.opam/default/.opam-switch/build/ocaml-freestanding.0.4.7/nolibc/include/sys/types.h:5:
# /home/asteen/.opam/default/lib/pkgconfig/../../include/solo5-bindings-hvt/crt/stdint.h:67:21: error: typedef redefinition with different types ('__uint64_t' (aka 'unsigned long long') vs 'unsigned long')
# typedef __uint64_t              uint64_t;
#                                 ^
# ./caml/config.h:114:26: note: previous definition is here
# typedef ARCH_UINT64_TYPE uint64_t;
#                          ^
# 2 warnings and 2 errors generated.
# gmake[1]: *** [Makefile:343: startup_nat_n.o] Error 1
# gmake[1]: Leaving directory '/home/asteen/.opam/default/.opam-switch/build/ocaml-freestanding.0.4.7/build/ocaml/runtime'
# gmake: *** [Makefile:76: build/ocaml/runtime/libasmrun.a] Error 2



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+- The following actions failed
| - build ocaml-freestanding 0.4.7
+-
```

**Comment by hannesm** [comment]
hi @adamsteen, this looks related to #63 -- and since #65 was merged (into master, but there's no release with these changes), I'd expect the build to work smoothly (at least for ocaml 4.08+). would you mind to try master instead of the released 0.4.7: if you `opam pin add ocaml-freestanding --dev`, this will clone and install master.

**Comment by adamsteen** [comment]
@hannesm `opam pin add ocaml-freestanding --dev` worked a treat, `/mirage-skeleton/tutorial/noop` and `/mirage-skeleton/applications/static_website_tls` worked nicely.

**Comment by mato** [comment]
> @hannesm `opam pin add ocaml-freestanding --dev` worked a treat, `/mirage-skeleton/tutorial/noop` and `/mirage-skeleton/applications/static_website_tls` worked nicely.

Thanks for testing this. It'll go into the next release, likely after OCaml
4.10 is released which should be in the next couple of weeks.

**Comment by mato** [comment]
This should be fixed in release 0.5.0 (in OPAM soon). Please re-open if not.
